IPython/core/tests/test_completer.py

```diff
@@ -841,47 +852,181 @@ def test_match_dict_keys(self):
         """
         delims = " \t\n`!@#$^&*()=+[{]}\\|;:'\",<>?"
 
-        keys = ["foo", b"far"]
-        assert match_dict_keys(keys, "b'", delims=delims) == ("'", 2, ["far"])
-        assert match_dict_keys(keys, "b'f", delims=delims) == ("'", 2, ["far"])
-        assert match_dict_keys(keys, 'b"', delims=delims) == ('"', 2, ["far"])
-        assert match_dict_keys(keys, 'b"f', delims=delims) == ('"', 2, ["far"])
-
-        assert match_dict_keys(keys, "'", delims=delims) == ("'", 1, ["foo"])
-        assert match_dict_keys(keys, "'f", delims=delims) == ("'", 1, ["foo"])
-        assert match_dict_keys(keys, '"', delims=delims) == ('"', 1, ["foo"])
-        assert match_dict_keys(keys, '"f', delims=delims) == ('"', 1, ["foo"])
+        def match(*args, **kwargs):
+            quote, offset, matches = match_dict_keys(*args, **kwargs)
+            return quote, offset, list(matches)
 
-        match_dict_keys
+        keys = ["foo", b"far"]
+        assert match(keys, "b'", delims=delims) == ("'", 2, ["far"])
+        assert match(keys, "b'f", delims=delims) == ("'", 2, ["far"])
+        assert match(keys, 'b"', delims=delims) == ('"', 2, ["far"])
+        assert match(keys, 'b"f', delims=delims) == ('"', 2, ["far"])
+
+        assert match(keys, "'", delims=delims) == ("'", 1, ["foo"])
+        assert match(keys, "'f", delims=delims) == ("'", 1, ["foo"])
+        assert match(keys, '"', delims=delims) == ('"', 1, ["foo"])
+        assert match(keys, '"f', delims=delims) == ('"', 1, ["foo"])
+
+        # Completion on first item of tuple
+        keys = [("foo", 1111), ("foo", 2222), (3333, "bar"), (3333, "test")]
+        assert match(keys, "'f", delims=delims) == ("'", 1, ["foo"])
+        assert match(keys, "33", delims=delims) == ("", 0, ["3333"])
+
+        # Completion on numbers
+        keys = [0xDEADBEEF, 1111, 1234, "1999", 0b10101, 22]  # 3735928559  # 21
```

Comment:
Dear black... https://github.com/psf/black/issues/379#issuecomment-449710601

```suggestion
        keys = [0xDEADBEEF, 1111, 1234, "1999", 0b10101, 22]  # 0xDEADBEEF = 3735928559; 0b10101 = 21
```